src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLoggingExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.Telemetry.Logging;
+
+/// <summary>
+/// Extensions for configuring logging.
+/// </summary>
+[Experimental]
+public static class ExtendedLoggingExtensions
+{
+    /// <summary>
+    /// Configure extended logging, enabling redaction and enrichment.
+    /// </summary>
+    /// <param name="services">The dependency injection container to add logging to.</param>
+    /// <param name="configure">Logging configuration options.</param>
+    /// <returns>The value of <paramref name="services"/>.</returns>
+    public static IServiceCollection AddExtendedLogging(this IServiceCollection services, Action<ILoggingBuilder> configure)
```

Comment:
This is experimental so I don't mind that much but I think this should be exposed on ILoggingBuilder instead of IServiceCollection.

Reply:
This is experimental only for the next few weeks. We will review the API and make it non-experimental before .NET 8 ships.

Why on logging builder instead of service collection? AddLogging is on service collection and this is a replacement for AddLogging, so it seemed reasonable to have this on the same interface.

Reply:
Because I would expect this to disappear when we merge these changes into the logger factory in .NET 9.

Reply:
So, before .NET 9 this will looke like this?
```csharp
services.AddLogging(x => x.AddExtendedLogging(/* ... */));
```

Reply:
Can you describe please how it will disappear in .NET 9? Will it be a breaking change?

Reply:
This method will be obsolete. The logger factory changed here will be integrated into the logger factory. These methods exists to support redaction and enrichment and shouldn’t be exposed